### ISSUE TITLE
Fix crash with null pointer when accessing session

### ISSUE THREAD

**Issue by dss-kenya** [snowplow/snowplow-android-tracker#581]
I recently upgraded to 4.1.1 from version 3.0.0 and there seems to be a crash reported from the sdk:

```
Caused by java.lang.NullPointerException: Attempt to invoke virtual method 'java.util.Map ee0.e.d()' on a null object reference
       at com.snowplowanalytics.snowplow.internal.session.Session.getSessionContext(Session.java:202)
       at com.snowplowanalytics.snowplow.internal.tracker.Tracker.workaroundForIncoherentSessionContext(Tracker.java:757)
       at com.snowplowanalytics.snowplow.internal.tracker.Tracker.track(Tracker.java:629)
       at com.snowplowanalytics.snowplow.internal.tracker.TrackerControllerImpl.track(TrackerControllerImpl.java:94)
```

i set the up the tracker as under:

```
Snowplow.setup(
            application,
            remoteConfig,
            listOf(configBundle)
        ) { configurationPair ->
            Snowplow.getDefaultTracker()?.attachGlobalContextToTracker() 
            -- manually attach global config since this is not possible at the moment with remoteConfig.
        }
```

and send the event as below:

```
Snowplow.getDefaultTracker()?.track(selfDescribingEvent)
```

Is there something i might be missing? Or may i please know why this crash is happening? I am confident this never happened with version 3.0.0

**Comment by mscwilson** [comment]
Thanks for the report! We will try to replicate it and get back to you soon.

**Comment by krobert** [comment]
HEy @mscwilson, we are experiencing the same issue, we were using 4.0.0 for a while, it started doing the issue, upgraded to 4.1.1 hoping it fixes it, but now there are more reports of this crash.
Its currently breaking the app, during start. Seems to be happening api 12-13 on newer devices like pixel 7.

**Comment by matus-tomlein** [comment]
Hi @krobert, thanks for reporting. We were able to replicate the error in PR #582 in case session tracking is paused and the tracker tracks an event. Do you by any chance pause session tracking in your app using `tracker.session.pause()`?

**Comment by krobert** [comment]
> .pause()

I updated my comment with the crash report from crashlytics.
No we are not pausing the session , we receive a crash at app start `onSessionUpdate` callback, when we are trying to get the session id from the tracker

**Comment by matus-tomlein** [comment]
Thanks @krobert, that's useful!

Just a couple more questions to help us identify the problem since I think yours is slightly different than the one reported previously:

1. Are you also using remote configuration to configure the tracker?
2. Is it true that you are calling the above snippet from inside the `onSessionUpdate` callback? In that case, why not use the `SessionState` object that is passed to the callback to retrieve the information? We will try to fix also the access from the `tracker` object, but using the state object passed to the callback can be a quick fix for the exception that you are getting.

**Comment by krobert** [comment]
@matus-tomlein 
1. yes, we configure the tracker right after, bit problematin, we have to do our own event caching mechanism 
2. sure we can do that

**Comment by matus-tomlein** [comment]
Hi @dss-kenya and @krobert, the bug has been fixed in the 4.1.2 release. I hope it will help, please let us know if not.

**Comment by dss-kenya** [comment]
Thank you @matus-tomlein 👍
